### DISCUSSION TITLE
Resetting Password with CLI tool

### DISCUSSION THREAD

**Discussion by kfreezen** [ory/kratos#2776] [Q&A]
Hey,

I have a use case where I want to reset the password for an admin account via a CLI tool (root only). Does kratos currently support such a thing?

If it's supported, what does the flow look like? As far as I can tell from the documentation, the farthest I can get the recovery/password reset flow to go is a link that is either emailed or copy-pasted.

If it's not supported, are there any plans to support such an automated flow in the future?

Where I've gotten so far is obtaining a session cookie and resetting the password, but instead of getting a 200 OK, I get back a 303 See Other.

A couple ways I could hack around this problem:
* spoof the `Referer` or other headers to make it look like my settings request is coming from the UI.
* Delete and re-register the admin user, which obliterates the old admin ID.

**Comment by vinckr** [comment]
>  reset the password for an admin account via a CLI tool (root only)

You can call the admin api and [create a recovery link](https://www.ory.sh/docs/reference/api#operation/adminCreateSelfServiceRecoveryLink) in the CLI tool (that has then to be opened in the browser). 

In the very near future (feature is already merged and being tested) you can use a short 6-digit code to reset the password. I am not sure, but maybe this could then be done without opening a browser. fyi @jonas-jonas 

Would be great if this is supported directly, but your hacks also seem OK to me.
